### DISCUSSION TITLE
Can't get it to work

### DISCUSSION THREAD

**Discussion by Tarpon907** [coddingtonbear/obsidian-web#155] [Q&A] [answered]
I'm trying to get Obsidian Web to work for the first time, but it isn't working.

I got the local rest API running and can use the interactive docs to access my vault. I configured the web extension with my API key.  I have also added my certificate to Chrome.

But when I try to clip a page, nothing happens.

Is something broken, or am I missing a step?

**Comment by coddingtonbear** [comment]
It's a little hard to make guesses with that little bit of information, but maybe I can help point you in the right direction after a few questions from you:

- When you're saying that you have the extension configured with your API key, are you saying that you see a green (or possibly orange) symbol next to the API key field?  If you see a red one, there's a problem for you to work out, and you should see an error message on the page somewhere.
- When you say "when I try to clip a page, nothing happens", what exactly do you mean:
   - A) You're clicking on the extension icon or pressing the keyboard shortcut in the chrome toolbar and it isn't displaying anything.
   - B) You're clicking on the extension icon or pressing the keyboard shortcut in the chrome toolbar, and although the extension window is shown, when you click on the "Send to Obsidian" button, nothing happens and the window remains displayed.
   - C) You're clicking on the extension icon or pressing the keyboard shortcut in the chrome toolbar, and although the extension window is shown, when you click on the "Send to Obsidian" button, the window disappears with a success message, but you can't find the note in your vault.

**Reply by Tarpon907** [reply]
* I have my API key in the API key field, and there's a green shield with a checkmark.  (Connected securely to the API with HTTPS).
* B.  The extension appears and I click on "Send to Obsidian."  The purple bar turns white.  The extension window remains displayed.  Nothing changes in obsidian.

**Reply by coddingtonbear** [reply]
Well, that case is luckily the easy one to diagnose -- if you open the chrome inspector from the menu in the upper-right corner, ("...", "More Tools", "Developer tools"), do any error messages appear when you click the "Send" button?

**Comment by Tarpon907** [comment]
The first "Failed to fetch" error and "Could not identify message" errors occur when I click the extension icon.  The second "Failed to fetch" error occurs when I click the send button.  I had your last comment highlighted when I did both actions.

![image](https://github.com/coddingtonbear/obsidian-web/assets/10379464/980ddd0d-f424-4b73-b7b2-9ce4aedb14e5)

**Reply by Tarpon907** [reply]
I got an error in the extension window that I have not seen before:

![image](https://github.com/coddingtonbear/obsidian-web/assets/10379464/0805922e-84e7-46b1-b432-b8592c9e4984)

**Reply by coddingtonbear** [reply]
"Could not identify message" is a red herring -- that'll appear if you open the pop-up repeatedly quickly; you can read more about that here: https://github.com/coddingtonbear/obsidian-web/discussions/134#discussioncomment-7557459.  In a similar vein, I think your "A listener indicated an asynchronous [...]" message is probably caused by a similar thing.

"Failed to fetch" really _sounds_ like it wasn't able to do the request at all, but it's unfortunately not enough information on its own.  I wonder if you could expand the "Received background request" shown immediately after you press "Send to Obsidian" in your log above to show what it was sending in the request.  Be sure to omit your API key, of course.

**Reply by Tarpon907** [reply]
It took a while before I could get that again.  I would hit the "Send to..." button in the extension window, and nothing would happen in the debug console.  It took a couple cycles of exiting Chrome and Obsidian and rebooting before I got anything in the debug console.

![image](https://github.com/coddingtonbear/obsidian-web/assets/10379464/e5878083-e9f2-441e-b2f0-158961939808)

**Reply by Tarpon907** [reply]
Oh, I missed the first bit.  Here.

![image](https://github.com/coddingtonbear/obsidian-web/assets/10379464/864c6542-d526-4610-b644-76bf0fdd4a71)

**Reply by coddingtonbear** [reply]
Hrm; I’m sort of perplexed right now since i’m not able to reproduce this kind of problem at the moment, but your request looks correct.

Before giving other suggestions; can you tell me details about the following:

- Your OS and its version
- Your browser and its version
- Obsidian’s version
- Obsidian Local REST API’s version

and confirm that you installed the package via the chrome web store?

**Reply by Tarpon907** [reply]
Windows 11
Chrome Version 120.0.6099.130
Obsidian 1.5.3
Local REST API 2.0.3

Yes, installed from the chrome web store.


Also, to fend off one more possible question, when I took the screenshots last night, Obsidian Web was the only extension I had enabled.  I disabled everything else and restarted Chrome (and rebooted).

**Reply by coddingtonbear** [reply]
I'm afraid I'm out of guesses at the moment, but it's true that I'm suspicious that there's something interesting happening in your personal environment since I use this every day right now, and then there's that `net::ERR_BLOCKED_BY_CLIENT` error in your console screenshot above, which although it isn't directly related to this extension, does point at _something_ blocking network requests.  That doesn't mean that it _isn't_ a bug, but until somebody else reports a problem, I have to assume it's something very specific to you.

**Reply by Tarpon907** [reply, marked as answer]
I finally got it to work, and I feel a little dumb.  But maybe this can serve as a warning to others.

I reset the Crypto, and then had to re-import the certificate into Chrome.  When I did, everything worked.  At that point, I found that the old certificate had been imported into the wrong section.    If it's not in "Trusted Root Certificate Authorities," it won't work.

**Reply by coddingtonbear** [reply]
I'm glad you figured it out; congrats!

**Reply by namsuu** [reply]
Thank you for the help.
As mentioned, we need to install the certificate on the correct path.
In my case, it generally works after specifying the Current User option selected for Store Location and the storage location as "Trusted Root Certification Authorities."

One more. All previously installed (wrong path) certificates must be deleted.
I am able to use it without any issues.

**Reply by coddingtonbear** [reply]
Oh, neat, if you've figured out how to do this on Chrome on Windows, I wonder if you'd mind updating the wiki page here: https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-get-my-browser-trust-my-Obsidian-Local-REST-API-certificate%3F to include a new section "# Chrome on Windows" with your instructions, @namsuu ?

**Reply by teknoviking** [reply]
@Tarpon907  This worked for me too after I went through Windows 11 certificate manager and deleted all the previous Obsidian certs that I had created in previous attempts to fix the error.  @coddingtonbear Thanks for all the great work on these add-ons!

**Reply by coddingtonbear** [reply]
Could I convince you into adding the steps you took [to the wiki section covering this topic](https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-get-my-browser-trust-my-Obsidian-Local-REST-API-certificate%3F), @teknoviking -- I do not personally use Windows, so I'd be a poor choice for writing instructions, and it sounds like you've figured out something that might be helpful for others!

**Reply by coddingtonbear** [reply]
Thanks @teknoviking for sending me the instructions; I've added them to the wiki now & hopefullly folks on Windows will now be able to get their certificates sorted out!
